common/djangoapps/student/views.py

```diff
@@ -1934,38 +1935,44 @@ def change_email_request(request):
 
     new_email = request.POST['new_email']
     try:
-        validate_email(new_email)
-    except ValidationError:
+        do_email_change_request(request.user, new_email)
+    except ValueError as err:
         return JsonResponse({
             "success": False,
-            "error": _('Valid e-mail address required.'),
-        })  # TODO: this should be status code 400  # pylint: disable=fixme
+            "error": err.message,
+        })
+    return JsonResponse({"success": True})
+
+
+def do_email_change_request(user, new_email, activation_key=uuid.uuid4().hex):
+    """
+    Given a new email for a user, does some basic verification of the new address and sends an activation message
+    to the new address. If any issues are encountered with verification or sending the message, a ValueError will
+    be thrown.
+    """
+    try:
+        validate_email(new_email)
+    except ValidationError:
+        raise ValueError(_('Valid e-mail address required.'))
+
+    if new_email == user.email:
+        raise ValueError(_('Old email is the same as the new email.'))
 
     if User.objects.filter(email=new_email).count() != 0:
         ## CRITICAL TODO: Handle case sensitivity for e-mails
```

Comment:
Did you consider trying to address this critical TODO? It can't be that critical since nothing has been done with it for a while. Maybe remove the comment if we don't think it is critical.